pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/src/main/java/org/apache/pinot/plugin/stream/kafka20/KafkaStreamMetadataProvider.java

```diff
@@ -72,6 +77,26 @@ public StreamPartitionMsgOffset fetchStreamPartitionOffset(OffsetCriteria offset
     }
   }
 
+  @Override
+  public PartitionLagInfo computePartitionLagInfo(ConsumerPartitionStatus currentConsumerStatus,
+      ConsumerPartitionStatus upstreamStatus) {
+    if (currentConsumerStatus != null && upstreamStatus != null) {
+      PartitionLagInfo partitionLagInfo = new PartitionLagInfo();
+      partitionLagInfo.setOffsetLag(computeOffsetLag(currentConsumerStatus.getMsgOffset(),
+          upstreamStatus.getMsgOffset()));
+      return partitionLagInfo;
+    } else {
+      LOGGER.warn("Unable to compute partition lag for {}", _topicPartition.toString());
+      return PartitionLagInfo.empty();
+    }
+  }
+
+  private String computeOffsetLag(StreamPartitionMsgOffset currentOffset,
```

Comment:
When the lag crosses a threshold it would be nice to see if this is due to a spike in messages or a slow backend. Is it possible to expose currentOffset and latestUpstreamOffset as counters?